### PR TITLE
Update apply-formatting.yml

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
## PR Summary

Adding formatting capability to python source files. Parthenon now makes use of several python scripts. It makes sense to expand automatic formatting to the python files. This is achieved in this pr by editing the formatting files that @AndrewGaspar implemented.

Now the ci will both look at python files and cpp files and run a pass or fail check. The par-hermes trigger can also be called as before. Which will auto format the python files in the repository. 

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [x] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
@par-hermes format

**Comment by JoshuaSBrown** [comment]
@par-hermes format

**Comment by JoshuaSBrown** [comment]
@par-hermes format

**Comment by JoshuaSBrown** [comment]
@par-hermes format

**Comment by JoshuaSBrown** [comment]
I don't think the Check C++ Formatting action is going to run for this pr because I have changed it.

**Comment by Yurlungur** [comment]
Nice! What about a command for local formatting? I.e., can we run `make format` to format the python scripts?

**Comment by JoshuaSBrown** [comment]
> Nice! What about a command for local formatting? I.e., can we run `make format` to format the python scripts?

Ya, I got lazy. Lol.

**Comment by Yurlungur** [comment]
> > Nice! What about a command for local formatting? I.e., can we run `make format` to format the python scripts?
> 
> Ya, I got lazy. Lol.

I vastly prefer doing local formatting over calling `par-hermes`, because I'm old-school and I like CLI's I guess. :P

**Comment by JoshuaSBrown** [comment]
> > > Nice! What about a command for local formatting? I.e., can we run `make format` to format the python scripts?
> > 
> > 
> > Ya, I got lazy. Lol.
> 
> I vastly prefer doing local formatting over calling `par-hermes`, because I'm old-school and I like CLI's I guess. :P

Making me earn my pay. :)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ cmake -DPARTHENON_LINT_DEFAULT=ON .
 ```
 
 ### Formatting Code
-We use clang-format to automatically format the code. If you have clang-format installed
+We use clang-format to automatically format the code cpp code. If you have clang-format installed
```
Comment (jlippuner) [inline CONTRIBUTING.md]:
```suggestion
We use clang-format to automatically format the C++ code. If you have clang-format installed
```

**Review by jlippuner — APPROVED** [review]
This is a great addition and I agree that we need automated formatting for Python code. I do agree that it would be nice to have local formatting of the Python code (like `make format`), although I don't consider it absolutely necessary. After all, we mostly work on the C++ code and the Python code doesn't change that often (and there isn't that much to begin with). So if it turns out be a huge pain to implement local Python formatting, I'd be ok to forgo it, provided that `par-hermes` works.

One more thing, this PR changes a lot of `.py` files, but it looks like just formatting. Did you make any code changes to any `.py` files? If so, I think we should have separate PRs for code and infrastructure changes and then all the formatting changes in another PR. But if all the changes to the Python files in this PR are purely formatting, then I think it's ok to have them in this PR.

**Comment by JoshuaSBrown** [comment]
> This is a great addition and I agree that we need automated formatting for Python code. I do agree that it would be nice to have local formatting of the Python code (like `make format`), although I don't consider it absolutely necessary. After all, we mostly work on the C++ code and the Python code doesn't change that often (and there isn't that much to begin with). So if it turns out be a huge pain to implement local Python formatting, I'd be ok to forgo it, provided that `par-hermes` works.
> 
> One more thing, this PR changes a lot of `.py` files, but it looks like just formatting. Did you make any code changes to any `.py` files? If so, I think we should have separate PRs for code and infrastructure changes and then all the formatting changes in another PR. But if all the changes to the Python files in this PR are purely formatting, then I think it's ok to have them in this PR.

No I did not change any py files, changes made are only to build configuration and ci.

**Comment by JoshuaSBrown** [comment]
> > > Nice! What about a command for local formatting? I.e., can we run `make format` to format the python scripts?
> > 
> > 
> > Ya, I got lazy. Lol.
> 
> I vastly prefer doing local formatting over calling `par-hermes`, because I'm old-school and I like CLI's I guess. :P

You should be happy now.

File: cmake/Format.cmake [hunk 2]
```diff
@@ -32,6 +35,16 @@ if (CLANG_FORMAT AND NOT CLANG_FORMAT_VERSION)
     endif()
 endif()
 
+if (BLACK)
+    # Get clang-format --version
```
Comment (pgrete) [inline cmake/Format.cmake]:
```suggestion
    # Get black --version
```

File: cmake/Format.cmake [hunk 3]
```diff
@@ -55,10 +68,37 @@ set(
     ${PROJECT_SOURCE_DIR}/example/[^\.]*.cpp ${PROJECT_SOURCE_DIR}/example/[^\.]*.hpp
 )
 
+# Specifically trying to exclude external here - I'm not sure if there's a better way
+set(
+    PY_GLOBS
+    ${PROJECT_SOURCE_DIR}/scripts/[^\.]*.py
+    ${PROJECT_SOURCE_DIR}/tst/[^\.]*.py
+    ${PROJECT_SOURCE_DIR}/example/[^\.]*.py
```
Comment (pgrete) [inline cmake/Format.cmake]:
```suggestion
    ${parthenon_SOURCE_DIR}/scripts/[^\.]*.py
    ${parthenon_SOURCE_DIR}/tst/[^\.]*.py
    ${parthenon_SOURCE_DIR}/example/[^\.]*.py
```

`PROJECT_SOURCE_DIR` depends on the latest `project()` call, so specifically using `parthenon_SOURCE_DIR` may be the more safe choice if Parthenon is included as subdirectory in a downstream code.

File: cmake/Format.cmake [hunk 4]
```diff
@@ -55,10 +68,37 @@ set(
...
+)
+
 if (CMAKE_VERSION VERSION_LESS "3.12.0")
     file(GLOB_RECURSE FORMAT_SOURCES ${GLOBS})
+    file(GLOR_RECURSE PY_FORMAT_SOURCES ${PY_GLOBS})
 else()
     file(GLOB_RECURSE FORMAT_SOURCES CONFIGURE_DEPENDS ${GLOBS})
+    file(GLOB_RECURSE PY_FORMAT_SOURCES CONFIGURE_DEPENDS ${PY_GLOBS})
 endif()
```
Comment (pgrete) [inline cmake/Format.cmake]:
```suggestion
file(GLOB_RECURSE FORMAT_SOURCES CONFIGURE_DEPENDS ${GLOBS})
file(GLOB_RECURSE PY_FORMAT_SOURCES CONFIGURE_DEPENDS ${PY_GLOBS})
```

Given the latest Kokkos submodule update the min. required CMake version is now > 3.12 so we don't need this conditional any more.

**Review by pgrete — APPROVED** [review]
Looks great. I also confirmed that `make format` locally works as expected!
Pending my minor comments I'm happy to see this going in.
Also, as next step, we could also consider adding a `flake8` step.

File: cmake/Format.cmake (hunk 3 above)
Comment (JoshuaSBrown) [reply cmake/Format.cmake]: Good catch.

**Comment by JoshuaSBrown** [comment]
Thanks everyone for the input.

**Review by Yurlungur — APPROVED** [review]
Awesome. Thanks, @JoshuaSBrown !

**Pull request merged**